library/classes/Document.class.php

```diff
@@ -695,6 +845,48 @@ function createDocument(
         return '';
     }
 
+    /**
+     * Retrieves the document data that has been saved to the filesystem or couch db.  If the $decrypt flag is set to
+     * true, it will return the unencrypted version of the data for the document.
+     * @param bool $decrypt True if the document should be decrypted, false otherwise
+     * @throws BadMethodCallException Thrown if the method is called when the document has been marked as deleted or expired
+     * @return false|string Returns false if the data failed to decrypt, or a string if the data decrypts or is unencrypted.
+     */
+    function get_data($decrypt = false)
+    {
+        $storagemethod = $this->get_storagemethod();
+
+        if ($this->has_expired()) {
+            throw new BadMethodCallException("Should not attempt to retrieve data from expired documents");
+        }
+        if ($this->is_deleted()) {
+            throw new BadMethodCallException("Should not attempt to retrieve data from deleted documents");
+        }
+
+        if ($storagemethod === self::STORAGE_METHOD_COUCHDB) {
+            // Taken from ccr/display.php
+            $couch_docid = $this->get_couch_docid();
+            $couch_revid = $this->get_couch_revid();
+            $couch = new CouchDB();
+            $resp = $couch->retrieve_doc($couch_docid);
+            $data = base64_decode($resp->data);
```

Comment:
there needs to be a decision point here for docs from couchdb:
1. If the document is encrypted, then decrypt it (or leave as is if the $force_no_decrypt is true)
2. If the document is not encrypted, then do the base64_decode